core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -0,0 +1,475 @@
+class AddNoteDialog : DialogFragment() {
+  private var zimFileName: String? = null
+  private var zimFileTitle: String? = null
+  private var articleTitle: String? = null
+
+  // Corresponds to "ArticleUrl" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
+  private lateinit var articleNoteFileName: String
+  private var noteFileExists = false
+  private var noteEdited = false
+
+  private lateinit var root: View
+
+  // Keeps track of state of the note (whether edited since last save)
+  // Stores path to directory for the currently open zim's notes
+  private var zimNotesDirectory: String? = null
+
+  @Inject
+  lateinit var sharedPreferenceUtil: SharedPreferenceUtil
+
+  @Inject
+  lateinit var zimReaderContainer: ZimReaderContainer
+
+  @JvmField @Inject
+  var alertDialogShower: AlertDialogShower? = null
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    coreComponent
+      .activityComponentBuilder()
+      .activity(requireActivity())
+      .build()
+      .inject(this)
+
+    // Returns name of the form ".../Kiwix/granbluefantasy_en_all_all_nopic_2018-10.zim"
+    zimFileName = zimReaderContainer.zimCanonicalPath
+    if (zimFileName != null) { // No zim file currently opened
+      zimFileTitle = zimReaderContainer.zimFileTitle
+      articleTitle = (activity as WebViewProvider?)?.getCurrentWebView()?.title
+
+      // Corresponds to "ZimFileName" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
+      val zimNoteDirectoryName = zimNoteDirectoryName
+      articleNoteFileName = getArticleNotefileName()
+      zimNotesDirectory = "$NOTES_DIRECTORY$zimNoteDirectoryName/"
+    } else {
+      onFailureToCreateAddNoteDialog()
+    }
+  }
+
+  private fun onFailureToCreateAddNoteDialog() {
+    context.toast(R.string.error_file_not_found, Toast.LENGTH_LONG)
+    closeKeyboard()
+    requireFragmentManager().beginTransaction().remove(this).commit()
+  }
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View {
+    super.onCreateView(inflater, container, savedInstanceState)
+    root = inflater.inflate(R.layout.dialog_add_note, container, false)
+    return root
+  }
+
+  private val zimNoteDirectoryName: String
+    get() {
+      val noteDirectoryName = getTextAfterLastSlashWithoutExtension(zimFileName!!)
+      return (if (noteDirectoryName.isNotEmpty()) noteDirectoryName else zimFileTitle)!!
+    }
+
+  private fun getArticleNotefileName(): String {
+    // Returns url of the form: "content://org.kiwix.kiwixmobile.zim.base/A/Main_Page.html"
+    val articleUrl = (activity as WebViewProvider?)?.getCurrentWebView()?.url
+    var noteFileName = ""
+    if (articleUrl == null) {
+      onFailureToCreateAddNoteDialog()
+    } else {
+      noteFileName = getTextAfterLastSlashWithoutExtension(articleUrl)
+    }
+    return (if (noteFileName.isNotEmpty()) noteFileName else articleTitle)!!
```

Comment:
please don't use the `!!` operator

Reply:
@gouri-panda

Reply:
done

Reply:
![image](https://user-images.githubusercontent.com/3358282/92363492-7a697800-f0e9-11ea-8b4e-9ce837180f9e.png)
Have you not pushed up a commit? Or is github freaking out on me

Reply:
done .sorry @macgills my bad.

Reply:
No problem, there have been a lot of requested changes, it can be difficult to keep them all straight